sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -272,52 +334,212 @@ public Read withUpToDateThreshold(Duration upToDateThreshold) {
 
       return input.apply(transform);
     }
+  }
+
+  /** Implementation of {@link #write}. */
+  @AutoValue
+  public abstract static class Write extends PTransform<PCollection<byte[]>, PDone> {
+    @Nullable
+    abstract String getStreamName();
+    @Nullable
+    abstract String getPartitionKey();
+    @Nullable
+    abstract KinesisPartitioner getPartitioner();
+    @Nullable
+    abstract Properties getProducerProperties();
+    @Nullable
+    abstract AWSClientsProvider getAWSClientsProvider();
+    abstract int getRetries();
+
+    abstract Builder builder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setStreamName(String streamName);
+      abstract Builder setPartitionKey(String partitionKey);
+      abstract Builder setPartitioner(KinesisPartitioner partitioner);
+      abstract Builder setProducerProperties(Properties properties);
+      abstract Builder setAWSClientsProvider(AWSClientsProvider clientProvider);
+      abstract Builder setRetries(int retries);
+      abstract Write build();
+    }
+
+    public Write withStreamName(String streamName) {
```

Comment:
Please add javadocs for the builder methods, explaining nontrivial aspects (e.g. which properties are required, which have a default value and what it is, etc.)

Reply:
Ok, sure